### PR TITLE
Changing schema

### PR DESCRIPTION
**Pull request by BenBeard115**
Editing schema and respective files to add user interactions.

### CODE REVIEW

File: database/schema.sql [hunk 1]
```diff
@@ -1,21 +1,45 @@
...
+);
+
+
+CREATE TABLE user_interaction(
+    interaction_id SERIAL PRIMARY KEY,
+    url_id INT NOT NONE,
+    type_id INT NOT NONE,
+    at TIMESTAMP NOT NONE,
+    FOREIGN KEY (url_id) REFERENCES url(url_id),
+    FOREIGN KEY (type_id) REFERENCES interaction_type(type_id)
```
Comment (OscarB28) [inline database/schema.sql:28]: new tables are good for more detail

File: dashboard/dashboard_functions.py [hunk 2]
```diff
@@ -72,6 +72,28 @@ def make_daily_archive_tracker_line(data: pd.DataFrame):
...
+
+    archived = alt.Chart(data).mark_line().encode(
+        x=alt.X("monthdate(at):O").title("Time"),
+        y=alt.Y("visit_count:Q").title("Archives Visited")).configure_line(color="#d15353")
+
+    st.altair_chart(archived, use_container_width=True)
+
+
+def make_daily_save_tracker_line(data: pd.DataFrame):
+    """Makes an daily archive tracker."""
```
Comment (OscarB28) [inline dashboard/dashboard_functions.py:87]: fix grammar

**Comment by BenBeard115** [comment]
Still needs to be combined with front end to upload interactions.

File: api/app.py [hunk 3]
```diff
@@ -147,13 +148,19 @@ def retrieve_searched_for_pages(input: str):
...
     """Uploads website information to the database."""
     connection = get_connection(environ)
     add_url(connection, response_data)
     add_website(connection, response_data)
 
 
+def upload_interaction_to_database(interaction_data: dict):
+    connection = get_connection(environ)
+    add_url(connection, interaction_data)
+    add_interaction(connection, interaction_data)
```
Comment (OscarB28) [inline api/app.py:161]: clean but needs docstring

File: api/upload_to_database.py [hunk 4]
```diff
@@ -123,9 +146,21 @@ def add_website(conn: extensions.connection, response_data: dict) -> None:
         'url': "https://www.youtube.co.uk",
         'html_filename': 'FAKE_HTML',
         'css_filename': 'FAKE_CSS',
-        'timestamp': datetime(2023, 6, 22, 19, 10, 20)
+        'screenshot_filename': 'FAKE_SCREENSHOT',
+        'scrape_timestamp': datetime(2023, 6, 22, 19, 10, 20),
+        'is_human': True,
+        'summary': 'FAKE SUMMARY',
+        'genre': 'media'
```
Comment (OscarB28) [inline api/upload_to_database.py:153]: good to have example data

**Review by OscarB28 — APPROVED** [review]

**Pull request merged**